modules/geo-layers/src/tile-layer/tile-layer.js

```diff
@@ -214,6 +225,22 @@ export default class TileLayer extends CompositeLayer {
       return tile.layers;
     });
   }
+
+  getPickObjects() {
+    const {deck} = this.context;
+    const width = deck.canvas.width;
+    const height = deck.canvas.height;
+    const layerIds = [this.id];
+    const x = 0;
+    const y = 0;
+    return deck.pickObjects({x, y, width, height, layerIds});
+  }
+
+  getRenderedFeatures() {
+    const {maxFeatures} = this.props;
+    const features = this.getPickObjects();
+    return maxFeatures ? features.splice(0, maxFeatures) : features;
```

Comment:
The maxFeatures prop here is not working as a stop gap for large feature sets. You are still spending all the CPU cycles building the array before truncating it.

Reply:
Yes... I thought about removing this property from the tile-layer and keeping it in the mvt-layer but I found it strange from the user's point of view. What do you think? remove it from tile-layer and keep it on in the mvt-layer?